DataCollector/src/main/java/com/braintreepayments/api/datacollector/DataCollector.java

```diff
@@ -32,8 +36,7 @@ public DataCollector(@NonNull Context context, @NonNull String authorization) {
         this(new BraintreeClient(context, authorization));
     }
 
-    @VisibleForTesting
-    DataCollector(@NonNull BraintreeClient braintreeClient) {
+    public DataCollector(@NonNull BraintreeClient braintreeClient) {
```

Comment:
I don't think we want these to be public - can we use the public methods instead wherever these are being used? Or use a library scope restriction (might require converting to kotlin)

Reply:
@saperi22 if it helps we can make these public and add `@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)` annotations until `DataCollector` is fully migrated to Kotlin. We may also want to suppress docs generation for these methods if we go that route.

Reply:
On second thought these should only be accessed by the corresponding test module so it should be fine as is. Android Studio may show it as an access error in the tests for now, but it will still compile.

I'd be curious to see where the error is though? It's good to call out–we may encounter this same issue a few times during the package renaming process because of the mismatch between Kotlin and Java access modifiers.

Reply:
> I'd be curious to see where the error is though?

https://github.com/braintree/braintree_android/blob/9ed08b7b363e66062281cc639e76014f3a75389c/LocalPayment/src/main/java/com/braintreepayments/api/LocalPaymentClient.java#L43
We are using it in another module -- which is throwing a compile error when I change the package name as the previous visibility modifier was default (package-private).

Reply:
Ok yeah I remember seeing these with some of the initial Kotlin refactor. Eventually they will need to be public when we move to Kotlin. The `@RestrictTo` annotation is our best tool to call out to the merchant that our test constructors are not part of the SDK's API.

Reply:
Does the restrictTo library scope annotation work if we convert this class to Kotlin and LocalPaymentClient is still in Java? Maybe we could have a fast follow PR to do that conversion since we want to eventually convert everything anyways

Otherwise, for the time being we could rely on doc strings to note that this constructor is for internal use only and not covered by semantic versioning - I know iOS does that sometimes - what do you think?

Reply:
It should work yeah I don't think `@RestrictTo` is specific to Kotlin. It's just a more explicit way of telling merchants not to use these methods that doesn't exist on iOS.

Reply:
I did some testing when starting these package name updates. `@RestrictTo` did work for Kotlin classes. So we should be good using the annotation.